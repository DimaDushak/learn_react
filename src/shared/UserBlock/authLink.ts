const authLink = `https://www.reddit.com/api/v1/authorize?client_id=${process.env.CLIENT_ID}&response_type=code&state=random_string&redirect_uri=http://localhost:3000/auth&duration=permanent&scope=read identity submit`;

export default authLink;
