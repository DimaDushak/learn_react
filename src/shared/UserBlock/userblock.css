.userBlock {
    align-items: center;
    display: flex;
}

.userBlockSizes img {
    height: 20px;
    margin-right: 7px;
    width: 20px;
}

.userBlockSizes span {
    font-size: 0.5rem;
    line-height: 0.6rem;
}

.authorAvatar {
    border-radius: 50%;
}

.anonim {
    color: var(--grey99);
}

@media (min-width: 1024px) {
    .userBlockSizes span {
        font-size: 0.7rem;
        line-height: 0.8rem;
    }
}
