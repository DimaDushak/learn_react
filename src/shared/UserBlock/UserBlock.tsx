import React from 'react';
import authLink from './authLink';
import styles from './userblock.css';

interface Props {
    link?: string;
    avatarSrc?: string;
    userName?: string;
    className?: string;
}

export default function UserBlock({
    link = authLink,
    avatarSrc = '/static/image/AnonimIcon.svg',
    userName,
    className = styles.userBlockSizes,
}: Props): JSX.Element {
    return (
        <a href={link} className={`${styles.userBlock} ${className}`}>
            <img src={avatarSrc} alt="#" className={styles.authorAvatar} />
            <span className={userName ? '' : styles.anonim}>
                {userName ?? 'Аноним'}
            </span>
        </a>
    );
}
