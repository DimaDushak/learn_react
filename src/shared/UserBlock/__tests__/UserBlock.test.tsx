import { render } from '@testing-library/react';
import React from 'react';
import UserBlock from '../UserBlock';

describe('UserBlock', () => {
    test('should render UserBlock without optional props (snapshot)', () => {
        const { container } = render(<UserBlock />);

        expect(container.firstChild).toMatchSnapshot();
    });

    test('should render UserBlock with optional props (snapshot)', () => {
        const { container } = render(
            <UserBlock
                link="#link"
                avatarSrc="#avatarSrc"
                userName="Дмитрий"
                className="testClass"
            />,
        );

        expect(container.firstChild).toMatchSnapshot();
    });
});
