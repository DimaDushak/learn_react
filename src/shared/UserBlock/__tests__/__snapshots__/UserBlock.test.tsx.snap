// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBlock should render UserBlock with optional props (snapshot) 1`] = `
<a
  class="userBlock testClass"
  href="#link"
>
  <img
    alt="#"
    class="authorAvatar"
    src="#avatarSrc"
  />
  <span
    class=""
  >
    Дмитрий
  </span>
</a>
`;

exports[`UserBlock should render UserBlock without optional props (snapshot) 1`] = `
<a
  class="userBlock userBlockSizes"
  href="https://www.reddit.com/api/v1/authorize?client_id=undefined&response_type=code&state=random_string&redirect_uri=http://localhost:3000/auth&duration=permanent&scope=read identity submit"
>
  <img
    alt="#"
    class="authorAvatar"
    src="/static/image/AnonimIcon.svg"
  />
  <span
    class="anonim"
  >
    Аноним
  </span>
</a>
`;
